tests/functional/container/test_container.py

```diff
@@ -787,6 +790,76 @@ def test_object_with_versioning_header_with_delete_many(self):
+        param_content = self.param_content(self.ref, path)
+        param_content['delete_marker'] = 1
+
+        # Versioning not enabled
+        self._create_content(path)
+        resp = self.request('POST', self.url_content('delete'),
+                            params=param_content)
+        self.assertEqual(400, resp.status)
+
+        # Versioning enabled
+        props = {"system": {"sys.m2.policy.version": '-1'}}
+        resp = self.request('POST', self.url_container('set_properties'),
+                            params=params_container,
+                            data=json.dumps(props))
+        self.assertEqual(200, resp.status)
+        resp = self.request('POST', self.url_content('delete'),
+                            params=param_content)
+        self.assertEqual(204, resp.status)
+
+        # Add delete marker to a delete marker
+        resp = self.request('POST', self.url_content('delete'),
+                            params=param_content)
+        self.assertEqual(400, resp.status)
+
+        # Add delete marker to a specific version
+        self._create_content(path, version=12345)
+        param_content['version'] = 12345
+        resp = self.request('POST', self.url_content('delete'),
+                            params=param_content)
+        self.assertEqual(204, resp.status)
+        resp = self.request('POST', self.url_content('get_properties'),
+                            params=param_content)
+        self.assertEqual(200, resp.status)
+        self.assertEqual(
+            '12345', resp.headers[OBJECT_METADATA_PREFIX + 'version'])
+        param_content['version'] = 12346
+        resp = self.request('POST', self.url_content('get_properties'),
+                            params=param_content)
+        self.assertEqual(200, resp.status)
+        self.assertEqual(
+            '12346', resp.headers[OBJECT_METADATA_PREFIX + 'version'])
+        print(resp.headers)
+        self.assertEqual(
+            'True', resp.headers[OBJECT_METADATA_PREFIX + 'deleted'])
+        del param_content['version']
+
+        # Add delete marker to a specific version with delete marker
+        param_content['version'] = 12345
+        resp = self.request('POST', self.url_content('delete'),
+                            params=param_content)
+        self.assertEqual(409, resp.status)
+        del param_content['version']
+
+    def test_object_with_versioned_objects_before_latest(self):
+        path = random_content()
+        params_container = self.param_ref(self.ref)
+
+        self._create_content(path)
+        props = {"system": {"sys.m2.policy.version": '-1'}}
+        resp = self.request('POST', self.url_container('set_properties'),
+                            params=params_container,
+                            data=json.dumps(props))
+        self.assertEqual(200, resp.status)
+
+        self._create_content(path, version=12345)
+        self._create_content(path, version=12345, create_status=409)
+
```

Comment:
Does it make sense to  test creation of a delete marker of object at version 12345 ?

Reply:
I already add a test for this case: https://github.com/open-io/oio-sds/pull/1841/commits/0b940c22ef5509f3eb7b8f6de2c1dde7fbc42665#diff-b840da077c2e6f9754bf0e5c70ebb72cR820